src/client/components/registerForm/index.js

```diff
@@ -1,13 +1,154 @@
 import React from 'react';
+import {Redirect} from 'react-router';
+import Validator from 'validator';
+
 import './index.scss';
+import Spinner from '../../../img/spinner.svg';
+import $api from '../../api/api.json';
+
+const MIN_PASSWORD_LENGTH = 6;
+const ANIMATION_SHAKING_DURATION = 500;
 
 class RegisterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      isLoading: false,
+      isInvalidFields: false,
+      isLoggedIn: false,
+      isRegistered: false,
+      isRegistrationFailure: false,
+      registrationFailureMessage: '',
+    };
+  }
+
+  isValidEmail(email) {
+    return Validator.isEmail(email);
+  }
+
+  isValidPassword(password) {
+    return password.length >= MIN_PASSWORD_LENGTH;
+  }
+
+  formHttpHeader(username, password) {
+    let myHeaders = new Headers();
+
+    myHeaders.append('Content-Type', 'application/json');
+    let myRequest = new Request($api.register, {
+      'method': 'POST',
+      'headers': myHeaders,
+      'body': JSON.stringify({'username': username, 'password': password}),
+    });
+
+    return myRequest;
+  }
+
+  registrationHttpRequest(username, password) {
+    let that = this;
+
+    that.setState({isLoading: true});
+    fetch(that.formHttpHeader(username, password)).then((response) => {
+      that.setState({
+        isRegistered: true,
+        isLoading: false,
+      });
+    }).catch((error) => {
+      that.setState({
+        isRegistrationFailure: true,
+        registrationFailureMessage: 'Registration failed!',
+      });
+    });
+  }
+
+  shakingAnimation() {
+    let that = this;
+
+    that.setState({isInvalidFields: true});
+    setTimeout(() => {
+      that.setState({isInvalidFields: false});
+    }, ANIMATION_SHAKING_DURATION);
+  }
+
+  doRegist(username, password) {
+    if (!this.isValidEmail(this.state.username)
```

Comment:
why does this function got the `username` and `password` as a parameter if we got it from the state?

also you could do `const {username, password} = this.state` then you don't have to use `this.state` every time (shorter lines)